lib/associations/belongs-to-many.js

```diff
@@ -298,6 +298,46 @@ var BelongsToMany = function(source, target, options) {
 
 util.inherits(BelongsToMany, Association);
 
+BelongsToMany.prototype.get = function(instances, options) {
+  var throughModel = this.throughModel;
+
+  var throughOptions = _.assign({
+    where: _.set({}, this.foreignKey, {$in: _.pluck(instances, this.source.primaryKeyAttribute)}),
+    logging: options.logging
+  });
+
+  var association = this;
+
+  return throughModel.findAll(throughOptions).then(function (throughs) {
+    var otherKey = association.otherKey
+      , targetIds = _.chain(throughs).pluck(otherKey).uniq().value();
+
+    var targetOptions = _.chain({
+      where: _.set({}, association.target.primaryKeyAttribute, {$in: targetIds})
+    })
+        .assign(association.target.$optClone(options))
+        .omit(['include'])
+        .value();
+
+    return Utils.Promise.join(throughs, _.isEmpty(targetIds) ? null : association.target.findAll(targetOptions));
```

Comment:
Yes, it's possible using a `JOIN` clause, but that would basically be like implementing a non-separated query on the hasMany between the Through model and the target model.
The major downside is that target objects that are referenced in multiple rows in the through table will be retrieved multiple times.